File: Backend/src/Meseta_Verda.Domain/Entities/SolicitudRepartidor.cs
using Meseta_Verda.Domain.Events;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.Contracts;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Meseta_Verda.Domain.Entities
{
   
    public class SolicitudRepartidor
    {
        [Key]
        public int IdSolicitud { get; set; }
        [Required]
        public int IdUsuario { get; set; }
        [ForeignKey("IdUsuario")]
        public Usuario Usuario { get; set; } = null!;
        // Aquí está el truco: se mapea como columna tipo jsonb
        [Column("datos_repartidor", TypeName = "jsonb")]
        public  DatosRepartidorDto DatosRepartidor { get; set; } = new();
        [StringLength(20)]
        public string Estado { get; set; } = "pendiente";

        public DateTime FechaSolicitud { get; set; } = DateTime.UtcNow;

       

    }
}

File: Frontend/agrotrade_frontend/lib/models/Repartidor/perfil/solicitud_repartidor.dart
import 'datos_repartidor.dart';

class SolicitudRepartidor {
  final int idSolicitud;
  final int idUsuario;
  final DatosRepartidor datosRepartidor;
  final String estado;
  final DateTime fechaSolicitud;

  const SolicitudRepartidor({
    required this.idSolicitud,
    required this.idUsuario,
    required this.datosRepartidor,
    this.estado = "pendiente",
    required this.fechaSolicitud,
  });

  factory SolicitudRepartidor.fromJson(Map<String, dynamic> json) {
    return SolicitudRepartidor(
      idSolicitud: json['idSolicitud'] as int,
      idUsuario: json['idUsuario'] as int,
      datosRepartidor: json['datos_repartidor'] != null 
          ? DatosRepartidor.fromJson(json['datos_repartidor'])
          : DatosRepartidor.fromJson({}),
      estado: json['estado'] as String? ?? 'pendiente',
      fechaSolicitud: json['fechaSolicitud'] != null
          ? DateTime.tryParse(json['fechaSolicitud'] as String) ?? DateTime.now()
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idSolicitud': idSolicitud,
      'idUsuario': idUsuario,
      'datos_repartidor': datosRepartidor.toJson(),
      'estado': estado,
      'fechaSolicitud': fechaSolicitud.toIso8601String(),
    };
  }
}
